### PR TITLE
chore: mev-rpc-related-improvements

### PR DESCRIPTION
**Pull request by Chef-Yogi**
---

## PR-Codex overview
This PR focuses on removing the dependency on the `useIsConnectedMetaMask` hook in favor of using the `useWalletClient` and enhancing the functionality around handling Ethereum chain support and MEV status.

### Detailed summary
- Removed `useIsConnectedMetaMask` from `MevSwapDetail.tsx` and `AddMevRpcButton.tsx`.
- Integrated `useWalletClient` to manage wallet interactions.
- Added `checkWalletSupportAddEthereumChain` function to verify if adding Ethereum chains is supported.
- Updated `fetchMEVStatus` to use `walletClient` instead of `window.ethereum`.
- Modified query keys in `useIsMEVEnabled` to include `walletClient`, `account`, and `chainId`.
- Introduced `useShouldShowMEVToggle` to determine visibility based on wallet support and MEV status.
- Enhanced error handling for wallet operations.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/web/src/views/Mev/hooks/index.ts [hunk 1]
```diff
@@ -76,7 +69,9 @@ export const useAddMevRpc = (onSuccess?: () => void, onBeforeStart?: () => void,
           console.info('RPC network added successfully!')
           onSuccess?.()
         } catch (error) {
-          console.error('Error adding RPC network:', error)
+          if ((error as any).code === -32601) {
```
Comment (chefjackson) [inline apps/web/src/views/Mev/hooks/index.ts:72]: we need to pre-detect before actually asking users to add the network?

Comment (chefjackson) [reply apps/web/src/views/Mev/hooks/index.ts:72]: also let's use wallet client instead of `window.ethereum`

Comment (Chef-Yogi) [reply apps/web/src/views/Mev/hooks/index.ts:72]: got it, let me make the adjustment

Comment (Chef-Yogi) [reply apps/web/src/views/Mev/hooks/index.ts:72]: > we need to pre-detect before actually asking users to add the network?

yeah, should have already done it in useIsMEVEnabled hook, let me move the error message to that hook

File: apps/web/src/views/Mev/hooks/index.ts [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
-import { useCallback, useMemo } from 'react'
+import { useCallback } from 'react'
+import { Address } from 'viem'
 import { bsc } from 'viem/chains'
+import { useWalletClient } from 'wagmi'
 
-async function fetchMEVStatus(): Promise<boolean> {
-  if (!window.ethereum || (!window.ethereum as any)?.request) {
+async function fetchMEVStatus(account: Address): Promise<{ mevEnabled: boolean; isError: boolean }> {
+  if (!window.ethereum || (!window.ethereum as any)?.request || account === '0x') {
```
Comment (chefjackson) [inline apps/web/src/views/Mev/hooks/index.ts:10]: we should use wallet client here as well?

Comment (Chef-Yogi) [reply apps/web/src/views/Mev/hooks/index.ts]: fixed

File: apps/web/src/views/Mev/hooks/index.ts [hunk 3]
```diff
@@ -1,59 +1,62 @@
...
+          data: '0x',
         },
       ],
     })
-    return result === '0x30'
+    return { mevEnabled: result === '0x30', isError: false }
   } catch (error) {
-    console.error('Error checking MEV status:', error)
-    return false
+    if ((error as any).code === -32601) console.error('wallet_addEthereumChain is not supported')
```
Comment (chefjackson) [inline apps/web/src/views/Mev/hooks/index.ts:31]: I don't quite understand the logic here. Isn't this request checking wether custom rpc endpoint is set up? Don't see why it has anything to do with `wallet_addEthereumChain`

Comment (Chef-Yogi) [reply apps/web/src/views/Mev/hooks/index.ts:31]: yeah, it seems the eth_call is enough to check, let me move this to when user actually click the add wallet button

**Review by chefjackson — APPROVED** [review]

**Pull request merged**